tests/fuzz/s2n_client_fuzz_test.c

```diff
@@ -0,0 +1,222 @@
+    "MIIBCAKCAQEAy1+hVWCfNQoPB+NA733IVOONl8fCumiz9zdRRu1hzVa2yvGseUSq\n"
+    "Bbn6k0FQ7yMED6w5XWQKDC0z2m0FI/BPE3AjUfuPzEYGqTDf9zQZ2Lz4oAN90Sud\n"
+    "luOoEhYR99cEbCn0T4eBvEf9IUtczXUZ/wj7gzGbGG07dLfT+CmCRJxCjhrosenJ\n"
+    "gzucyS7jt1bobgU66JKkgMNm7hJY4/nhR5LWTCzZyzYQh2HM2Vk4K5ZqILpj/n0S\n"
+    "5JYTQ2PVhxP+Uu8+hICs/8VvM72DznjPZzufADipjC7CsQ4S6x/ecZluFtbb+ZTv\n"
+    "HI5CnYmkAwJ6+FSWGaZQDi8bgerFk9RWwwIBAg==\n"
+    "-----END DH PARAMETERS-----\n";
+
+static int MAX_NEGOTIATION_ATTEMPTS = 10;
+
+int buffer_read(void *io_context, uint8_t *buf, uint32_t len)
+{
+    struct s2n_stuffer *in_buf;
+    int n_read, n_avail;
+
+    if (buf == NULL) {
+        return 0;
+    }
+
+    in_buf = (struct s2n_stuffer *) io_context;
+    if (in_buf == NULL) {
+        errno = EINVAL;
+        return -1;
+    }
+
+    // read the number of bytes requested or less if it isn't available
+    n_avail = s2n_stuffer_data_available(in_buf);
+    n_read = (len < n_avail) ? len : n_avail;
+
+    if (n_read == 0) {
+        errno = EAGAIN;
+        return -1;
+    }
+
+    s2n_stuffer_read_bytes(in_buf, buf, n_read);
+    return n_read;
+}
+
+int buffer_write(void *io_context, const uint8_t *buf, uint32_t len)
+{
+    return len;
+}
+
+static struct s2n_config *client__config;
+
+static void s2n_server_fuzz_atexit()
+{
+    s2n_config_free(client__config);
+    s2n_cleanup();
+}
+
+int LLVMFuzzerInitialize(const uint8_t *buf, size_t len)
+{
+    GUARD(s2n_init());
+    GUARD(atexit(s2n_server_fuzz_atexit));
+    GUARD(setenv("S2N_ENABLE_CLIENT_MODE", "1", 0));
+
+    /* Set up Server Config */
+    notnull_check(client__config = s2n_config_new());
+    GUARD(s2n_config_add_cert_chain_and_key(client__config, certificate_chain, private_key));
+    GUARD(s2n_config_add_dhparams(client__config, dhparams));
+
+    return 0;
+}
+
+int LLVMFuzzerTestOneInput(const uint8_t *buf, size_t len)
+{
+    if(len < S2N_TLS_RECORD_HEADER_LENGTH){
+        return 0;
+    }
+
+    struct s2n_stuffer in;
+    GUARD(s2n_stuffer_alloc(&in, len));
+    GUARD(s2n_stuffer_write_bytes(&in, buf, len));
+
+    /* Set up Server Connection */
+    struct s2n_connection *client_conn;
+    notnull_check(client_conn = s2n_connection_new(S2N_CLIENT));
+    GUARD(s2n_connection_set_recv_cb(client_conn, &buffer_read));
+    GUARD(s2n_connection_set_send_cb(client_conn, &buffer_write));
+    GUARD(s2n_connection_set_recv_ctx(client_conn, &in));
+    GUARD(s2n_connection_set_config(client_conn, client__config));
+    GUARD(s2n_connection_set_blinding(client_conn, S2N_SELF_SERVICE_BLINDING));
+    client_conn->delay = 0;
+
+    /* Let Server receive data and attempt Negotiation */
+    int num_attempted_negotiations = 0;
+    s2n_blocked_status client_blocked;
+    do {
+        s2n_negotiate(client_conn, &client_blocked);
+        num_attempted_negotiations += 1;
+    } while(!client_blocked && num_attempted_negotiations < MAX_NEGOTIATION_ATTEMPTS);
+
+    /* Clean up */
+
```

Comment:
Do we see much extra branch coverage if we add an s2n_shutdown() call here? It will replicate what real client code looks like. We need not call it repeatedly since it waits for a close_notify from the peer(which will probably never come).